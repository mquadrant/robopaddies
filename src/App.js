import React ,{useState,useEffect}from 'react';
import {connect} from 'react-redux';
import CardList from './components/CardList'
import SearchBar from './components/SearchBar'
import Scroll from './components/Scroll'
import ErrorBoundry from './components/ErrorBoundry';

import {requestRobots} from './redux/action';

const mapStateToProps = state => {
  return {
    robots: state.requestRobots.robots,
    isPending: state.requestRobots.isPending
  }
}

const mapDispatchToProps = dispatch => {
 return {
  requestForRobots: () => dispatch(requestRobots())
}
}

function App(props) {
  const {requestForRobots,robots} = props;
  const [filteredRobots, setFilteredRobots] = useState([]);
  
  useEffect(()=>{ 
    requestForRobots();
  },[requestForRobots])

  useEffect(()=>{ 
    setFilteredRobots(robots);
  },[robots])

  const searchHandler = (e)=>{ 
    filterRobots(e.target.value);
  }

  const filterRobots = (term)=>{
    let re = new RegExp(term, "gi");
    setFilteredRobots(robots.filter(robot=>{
      return re.test(robot.name) || re.test(robot.email)
    }))
  }
  
  return (
          <div className="app-content">
            <div className="text-center my-12">
            <h1 className="text-5xl my-8">ROBOPADDIES</h1>
            <SearchBar searchChange={searchHandler}/>
            </div>
          <hr className="border-green-200"/>
          {setFilteredRobots.length?(
          <Scroll>
            <ErrorBoundry>
              <CardList robots={filteredRobots}/>
            </ErrorBoundry>
          </Scroll>):null}
        </div>
  )
    
}

export default connect(mapStateToProps, mapDispatchToProps)(App);