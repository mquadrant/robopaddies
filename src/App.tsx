import React from 'react';
import fetch from 'node-fetch'
import {connect} from 'react-redux';
import MainPage from './containers/MainPage';

import {requestRobots} from './redux/action';

const mapStateToProps = (state:any) => {
  return {
    robots: state.requestRobots.robots,
    isPending: state.requestRobots.isPending
  }
}

const mapDispatchToProps = (dispatch:any )=> {
 return {
  requestForRobots: () => dispatch(requestRobots(fetch))
}
}

function App(props:any) {
  return (
    <MainPage {...props}/>
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
