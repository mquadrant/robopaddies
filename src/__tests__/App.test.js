import React from 'react';
import {Provider} from 'react-redux';
import { mount} from 'enzyme';
import App from '../App';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const defaultStore = {requestRobots:{robots: [] }};
const mockedStore = configureMockStore([thunk])(defaultStore);

export const mountWithProvider = children => (store = mockedStore) =>
  mount(<Provider store={store}>{children}</Provider>);



it('renders app without crashing', () => {
    const props = {};
    // const myCustomState = {
        
    //   };
const wrapper = mountWithProvider(<App {...props} />)();
expect(wrapper.exists()).toBe(true);
  expect(wrapper).toMatchSnapshot();
});
