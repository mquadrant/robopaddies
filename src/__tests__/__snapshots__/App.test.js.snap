// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)>
    <App
      requestForRobots={[Function]}
      robots={Array []}
    >
      <MainPage
        requestForRobots={[Function]}
        robots={Array []}
      >
        <div
          className="app-content"
        >
          <div
            className="text-center my-12"
          >
            <h1
              className="text-5xl my-8"
            >
              ROBOPADDIES
            </h1>
            <SearchBar
              searchChange={[Function]}
            >
              <div>
                <input
                  className="shadow appearance-none border rounded w-36 py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search Robots"
                  type="text"
                />
              </div>
            </SearchBar>
          </div>
          <hr
            className="border-green-200"
          />
        </div>
      </MainPage>
    </App>
  </Connect(App)>
</Provider>
`;
